---
title: "Other Resources"
teaching: 15
exercises: 0
questions:
- "Where can I find other metagenomic resources?"
objectives:
- "Know some places where to find more information."
keypoints:
- "Metagenomes are complicated to analyse. Luckily, there are lots of tools available!"
- "The best tools to use change frequently. Do your research when looking for the most suitable tools to use for your analysis."
---

## Other resources
We hope that you have found these lessons useful. However, this course is only the start! There is lots more you can learn about analysing and understanding metagenomics, and how to apply these techniques to your own dataset.

A quick roundup of useful learning resources:
- This [Phyloseq tutorial](https://joey711.github.io/phyloseq/) contains many more examples of metagenome data manipulation. 
- A'nvio developer Meren has a blog entry about [History of metagenomics](http://merenlab.org/2020/07/27/history-of-metagenomics/)
- The same developer also has several videos explaining topics such as [Metapangenomics: A nexus between pangenomes and metagenomes](https://youtu.be/C3fHlccFxJw), [The power of metagenomic read recruitment](https://youtu.be/MqD4aN1p1qA) and [Genome-resolved metagenomics: key concepts in reconstructing genomes from metagenomes](https://youtu.be/RjNdHGK4ruo). 

There are also some great non-command line resources out there that may be great to get you started. Here are some of our suggestions, although this is by no means an exhaustive list.

### MG-RAST

A useful and easy to use resource is [MG-RAST](https://www.mg-rast.org/), an online metagenomic platform where you can upload your raw data with its corresponding metadata and get a full taxonomic analysis of it. MG-RAST is also a big repository of available data for future experiments. 

A downside of this resource is that it is not possible to modify the steps and parameters in the MG-RAST workflow, meaning there is less flexibility for implementing your preferred analysis tools.

Here is an example of some of the figures generated by MG-RAST. It annotates the metabolic function and can give you an idea of what your organisms/ individual MAGs are capable of metabolising.

<a href="../fig/03-11-02.png">
  <img src="../fig/03-11-02.png" alt="Pie chart showing the relative abundance at phylum level, and the legend with the phylum names, read count and percentages." />
</a>

<a href="../fig/03-11-04.png">
  <img src="../fig/03-11-04.png" alt="Pie chart showing the relative abundance of general functional categories, and the legend with the category names, read count and percentages." />
</a>

<a href="../fig/03-11-05.png">
  <img src="../fig/03-11-05.png" alt="Pie chart showing the relative abundance of specific functional categories, and the legend with the category names, read count and percentages." />
</a>

MG-RAST has is a very useful tool for taking a quick look at your data, as well as storing and sharing it.

### Krona for visualization

[Krona](https://github.com/marbl/Krona/wiki) is a hierarchical
data visualisation software. It allows data to be explored with zooming,
multi-layered pie charts and includes support for several bioinformatics
tools and raw data formats. Using Krona you can generate HTML reports on the taxonomy in your metagenome like the one below.

<a href="{{ page.root }}/fig/03-06-03.png">
  <img src="{{ page.root }}/fig/03-06-03.png" alt="Krona displays a circled-shape bacterial taxonomy plot with abundance percentages of each taxa " />
</a>

### iTOL, the interactive tree of life

[iTOL](https://itol.embl.de/) is a free website which can be used to generate publication quality phylogenetic trees. Below shows the interface you can use to customise parts of the plot. There are also several [video tutorials](https://itol.embl.de/video_tutorial.cgi) to show you how to get started. You can also share your trees with other people easily. 

This tool is useful if you want to make a tree based on the 16S sequence for several genomes in your metagenome (annotated by Prokka), or if you are interested in a particular set of genes within a genome.

<a href="{{ page.root }}/fig/iTOL.png">
  <img src="{{ page.root }}/fig/iTOL.png" alt="iTOL displays a circular plot with the phylogeny of different bacteria, eukaryotes and archaea, representing the tree of life  " />
</a>

### KAIJU - alternative classification tool
Kraken2 (the tool we used) is a very popular command line tool for annotating the short reads in your metagenome. KAIJU is another popular tool which has both a [command line version](https://github.com/bioinformatics-centre/kaiju) and a [web based version](https://kaiju.binf.ku.dk/server). It has databases for microbial eukaryotes and viruses as well as bacterial and archaeal refseq genomes. 

Here's what the web version looks like:

<a href="{{ page.root }}/fig/KAIJU_website.png">
  <img src="{{ page.root }}/fig/KAIJU_website.png" alt="iTOL displays a circular plot with the phylogeny of different bacteria, eukaryotes and archaea, representing the tree of life  " />
</a>

### MetaPhlAn for taxonomic profiling

[MetaPhlAn 4](http://segatalab.cibio.unitn.it/tools/metaphlan/index.html) is another very popular tool for profiling taxonomy in microbial communities. The latest associated paper is [here](https://www.biorxiv.org/content/10.1101/2022.08.22.504593v1). 

One benefit of MetaPhlAn is that it has an actively updated [github repository](https://github.com/biobakery/MetaPhlAn/) as well as an active support [forum](https://forum.biobakery.org/c/microbial-community-profiling/metaphlan/7).It was developed as part of the [BioBakery](https://huttenhower.sph.harvard.edu/tools/), which also has many other excellent tools and resources.

### Identifying functional domains with InterProScan

If you are interested in identifying the functional domain of proteins, InterProScan should be the first tool you use. It has both a [command line version](https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html) for annotating many proteins at once, and a [website version](http://www.ebi.ac.uk/interpro/search/sequence/) for predicting smaller numbers of proteins. It produces an exportable graphical output as shown below.

<a href="{{ page.root }}/fig/interproscan.png">
  <img src="{{ page.root }}/fig/interproscan.png" alt="Screenshot from InterProScan which shows domains that have been identified in a protein sequence " />
</a>

> ## Recommended reading:
> Review articles can be a useful source of information and some contain comprehensive lists of different tools and databases to use for your metagenomic analysis. Here are two we recommend:
> - [TaÅŸ et al.,2021](https://www.sciencedirect.com/science/article/pii/S0958166921000240)
> - [Yang et al., 2021](https://www.sciencedirect.com/science/article/pii/S2001037021004931#s0055)
{: .callout}

{% include links.md %}
