---
title: "Other Resources"
teaching: 5
exercises: 0
questions:
- "Where are other metagenomic resources?"
objectives:
- "Know some places where to find more information."
keypoints:
- "Metagenomes are complicated to analyse, there are lots of tools available, and the best ones to use change frequently. Do you research and look for what are the most suitable tools to use for your analysis."
---

## Other resources
We hope that you have found these lessons useful, however it is only the start to analysing and understanding metagenomics, and how to use this with your own dataset.
The [Phyloseq tutorial](https://joey711.github.io/phyloseq/) contains many more examples of metagenome data manipulation. A'nvio developer Meren has a blog entry about [History of metagenomics](http://merenlab.org/2020/07/27/history-of-metagenomics/) and several videos explaining [Metapangenomics: A nexus between pangenomes and metagenomes](https://youtu.be/C3fHlccFxJw), [The power of metagenomic read recruitment](https://youtu.be/MqD4aN1p1qA),[Genome-resolved metagenomics: key concepts in reconstructing genomes from metagenomes](https://youtu.be/RjNdHGK4ruo. There are also some great non command line resources out there that may be great to get you started. This is not an exhaustive list but are some of our suggestions.

### MG-RAST

A useful and easy to use resource is [MG-RAST](https://www.mg-rast.org/), an online metagenomic platform where you can upload your raw data with its corresponding metadata and get a full taxonomic analysis of it. MG-RAST is a great place to get started in this type of analyzes and it is also a big repository of available data for future experiments. On the downside, it is not possible to greatly modify the steps and parameters in the MG-RAST workflow, so there is not much room when it comes to implementing our preferred analysis tools when using MG-RAST.

Here is an example of some of the figures that get generated by MG-RAST. It annotates the metabolic function and can give you an idea of what your organisms/ individual MAGs are capable of meabolising.

<a href="../fig/03-11-02.png">
  <img src="../fig/03-11-02.png" alt="Pie chart showing the relative abundance at phylum level, and the legend with the phylum names, read count and percentages." />
</a>



<a href="../fig/03-11-04.png">
  <img src="../fig/03-11-04.png" alt="Pie chart showing the relative abundance of general functional categories, and the legend with the category names, read count and percentages." />
</a>

<a href="../fig/03-11-05.png">
  <img src="../fig/03-11-05.png" alt="Pie chart showing the relative abundance of specific functional categories, and the legend with the category names, read count and percentages." />
</a>

MG-RAST has it's own specific pipeline, so it is a very useful tool to have a quick look of your data, and also to store it and share it!

### Krona for visualization

[Krona](https://github.com/marbl/Krona/wiki) is a hierarchical
data visualization software. Krona allows data to be explored with zooming,
multi-layered pie charts and includes support for several bioinformatics
tools and raw data formats. It allows you to generate html reports on the taxonomy in your metagenome, an example is shown beneath.

<a href="{{ page.root }}/fig/03-06-03.png">
  <img src="{{ page.root }}/fig/03-06-03.png" alt="Krona displays a circled-shape bacterial taxonomy plot with abundance percentages of each taxa " />
</a>

### iTOL the interactive tree of life

[iTOL](https://itol.embl.de/) is a free website which can be used to generate publication quality phylogenetic trees. Below shows the tree of life and the flexibility in the parts of the plot that you can edit. There are also several [video tutorials](https://itol.embl.de/video_tutorial.cgi) to show you how to get started. You can also share your trees with other people easily. This would be useful if you wanted to make a tree based on the 16S sequence for several genomes in your metagenome, which will be annotated by prokka, or if you are interested in a particular set of genes within a genome.

<a href="{{ page.root }}/fig/03-06-03.png">
  <img src="{{ page.root }}/fig/iTOL.png" alt="iTOL displays a circular plot with the phylogeny of different bacteria, eukaryotes and archaea, representing the tree of life  " />
</a>

> ## Recommended reading:
> These two reviews [Ta≈ü et al.,2021](https://www.sciencedirect.com/science/article/pii/S0958166921000240) [Yang et al., 2021](https://www.sciencedirect.com/science/article/pii/S2001037021004931#s0055) have comprehensive lists of different tools and databases to use for your metagenomic analysis.

{: .callout}

{% include links.md %}
