---
title: "Other Resources"
teaching: 15
exercises: 0
questions:
- "Where are other metagenomic resources?"
objectives:
- "Know some places where to find more information."
keypoints:
- "Metagenomes are complicated to analyse, there are lots of tools available, and the best ones to use change frequently. Do you research and look for what are the most suitable tools to use for your analysis."
---

## Other resources
We hope that you have found these lessons useful. However, this course is only the start! There is lots more you can learn about analysing and understanding metagenomics, and how to apply these techniques to your own dataset.

A quick roundup of useful learning resources:
- This [Phyloseq tutorial](https://joey711.github.io/phyloseq/) contains many more examples of metagenome data manipulation. 
- A'nvio developer Meren has a blog entry about [History of metagenomics](http://merenlab.org/2020/07/27/history-of-metagenomics/)
- The same developer also has several videos explaining [Metapangenomics: A nexus between pangenomes and metagenomes](https://youtu.be/C3fHlccFxJw), [The power of metagenomic read recruitment](https://youtu.be/MqD4aN1p1qA) and [Genome-resolved metagenomics: key concepts in reconstructing genomes from metagenomes](https://youtu.be/RjNdHGK4ruo). 

There are also some great non command line resources out there that may be great to get you started. This is not an exhaustive list but are some of our suggestions.

### MG-RAST

A useful and easy to use resource is [MG-RAST](https://www.mg-rast.org/), an online metagenomic platform where you can upload your raw data with its corresponding metadata and get a full taxonomic analysis of it. MG-RAST is also a big repository of available data for future experiments. 

A downside of this resource is that it is not possible to modify the steps and parameters in the MG-RAST workflow, meaning there is less flexibility for implementing your preferred analysis tools.

Here is an example of some of the figures generated by MG-RAST. It annotates the metabolic function and can give you an idea of what your organisms/ individual MAGs are capable of metabolising.

<a href="../fig/03-11-02.png">
  <img src="../fig/03-11-02.png" alt="Pie chart showing the relative abundance at phylum level, and the legend with the phylum names, read count and percentages." />
</a>

<a href="../fig/03-11-04.png">
  <img src="../fig/03-11-04.png" alt="Pie chart showing the relative abundance of general functional categories, and the legend with the category names, read count and percentages." />
</a>

<a href="../fig/03-11-05.png">
  <img src="../fig/03-11-05.png" alt="Pie chart showing the relative abundance of specific functional categories, and the legend with the category names, read count and percentages." />
</a>

MG-RAST has is a very useful tool for taking a quick look at your data, as well as storing and sharing it.

### Krona for visualization

[Krona](https://github.com/marbl/Krona/wiki) is a hierarchical
data visualization software. It allows data to be explored with zooming,
multi-layered pie charts and includes support for several bioinformatics
tools and raw data formats. It allows you to generate HTML reports on the taxonomy in your metagenome like the one below.

<a href="{{ page.root }}/fig/03-06-03.png">
  <img src="{{ page.root }}/fig/03-06-03.png" alt="Krona displays a circled-shape bacterial taxonomy plot with abundance percentages of each taxa " />
</a>

### iTOL the interactive tree of life

[iTOL](https://itol.embl.de/) is a free website which can be used to generate publication quality phylogenetic trees. Below shows the tree of life and the flexibility in the parts of the plot that you can edit. There are also several [video tutorials](https://itol.embl.de/video_tutorial.cgi) to show you how to get started. You can also share your trees with other people easily. This would be useful if you wanted to make a tree based on the 16S sequence for several genomes in your metagenome, which will be annotated by prokka, or if you are interested in a particular set of genes within a genome.

<a href="{{ page.root }}/fig/iTOL.png">
  <img src="{{ page.root }}/fig/iTOL.png" alt="iTOL displays a circular plot with the phylogeny of different bacteria, eukaryotes and archaea, representing the tree of life  " />
</a>

### KAIJU alternative classification tool for metagenomics
Kraken2 is a very popular command line tool for annotating the short reads in your metagenome, however KAIJU is another popular alternative that has both a [command line version](https://github.com/bioinformatics-centre/kaiju) and a [web based version](https://kaiju.binf.ku.dk/server) you can run jobs from. It has databases for microbial eukaryotes and viruses as well as bacterial and archaeal refseq genomes. The website version looks as shown below

<a href="{{ page.root }}/fig/KAIJU_website.png">
  <img src="{{ page.root }}/fig/KAIJU_website.png" alt="iTOL displays a circular plot with the phylogeny of different bacteria, eukaryotes and archaea, representing the tree of life  " />
</a>

### MetaPhlAn for taxonomic profiling

[MetaPhlAn 4](http://segatalab.cibio.unitn.it/tools/metaphlan/index.html) is another very popular tool for profiling taxonomy in microbial communities. The latest associated paper is [here](https://www.biorxiv.org/content/10.1101/2022.08.22.504593v1). One benefit of MetaPhlAn is that it has an actively updated [github repository](https://github.com/biobakery/MetaPhlAn/), and there is active support [forum](https://forum.biobakery.org/c/microbial-community-profiling/metaphlan/7) from the people who have developed it as part of the [BioBakery](https://huttenhower.sph.harvard.edu/tools/), which also has many other excellent tools and resources available.

### Identifying functional domains with InterProScan

If you are interested in identifying the functional domain of proteins of interest, InterProScan should be the first tool to use. This has both a [command line version](https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html) for annotating many proteins at once, and a website version for predicting individual, or a small number of proteins. This will give you a graphic output you can export as shown below

<a href="{{ page.root }}/fig/interproscan.png">
  <img src="{{ page.root }}/fig/interproscan.png" alt="Screenshot from InterProScan which shows domains that have been identified in a protein sequence " />
</a>


> ## Recommended reading:
> These two reviews [Ta≈ü et al.,2021](https://www.sciencedirect.com/science/article/pii/S0958166921000240)  and [Yang et al., 2021](https://www.sciencedirect.com/science/article/pii/S2001037021004931#s0055) have comprehensive lists of different tools and databases to use for your metagenomic analysis.

{: .callout}

{% include links.md %}
